src/modules/cell.js

```diff
@@ -198,11 +198,13 @@
       if (width < tempWidth) {
         width = tempWidth;
       }
-      if (options.maxTextLength && width !== 0) {
-        amountOfLines = Math.cell(width / options.maxTextLength)
-      } else if (width !== 0) {
-        amountOfLines = text.length;
-      }
+    }
+
+    if (options.maxWidth){
+      amountOfLines = this.splitTextToSize(text, options.maxWidth)
```

Comment:
`.length` is missing ;)

We should call `splitTextToSize` before the loop so the width returned is the actual width of the text (might be smaller than `maxWidth`).

Reply:
You're right! I've added the missing `.length` and moved the `splitTextToSize` call up above the loop. I've also updated the tests such that we have:

* A width and height test without maxWidth using `splitTextToSize` before calling getTextDimensions in order to test correct multiline handling without `maxWidth`
* A width and height test with just plain text and withou `maxWidth`
* A width and height test with `maxWidth` set to a number larger than the actual text width
* A width and height test with `maxWidth` set to a number smaller than the actual text width
* I've kept the width and height tests for the empty string and the array with an empty string

I think these cases ought to cover the relevant use cases :)